### PR TITLE
config: enable format-checker for tikv/pd and require commit message with issue number

### PR DESCRIPTION
**Pull request by Mini256**
Enable the format checker for **tikv/tikv** and **tikv/pd** repo:

The format checker will check the message part of all commit in the PR. If none of the commit messages contains the associated issue number, the robot will add a new comment to prompt the contributor to add it. At the same time, the robot will add the label `do-not-merge/invalid-commit-message` for PR to prevent PR from merging

Test cases:

- https://github.com/ti-community-infra/test-dev/pull/468
- https://github.com/ti-community-infra/test-dev/pull/469
- https://github.com/ti-community-infra/test-dev/pull/470

Refer document: 

https://book.prow.tidb.io/#/plugins/format-checker

### CODE REVIEW

File: prow/config/plugins.yaml [hunk 1]
```diff
@@ -784,6 +788,10 @@ external_plugins:
     - name: ti-community-contribution
       events:
         - pull_request
+    - name: ti-community-format-checker
+      events:
+        - pull_request
+        - issues
```
Comment (tisonkun) [inline prow/config/plugins.yaml:794]: No issues?

**Comment by tisonkun** [comment]
cc @BusyJay @rleungx @disksing please confirm this config for requiring PR issue association on tikv/tikv and tikv/pd .

cc @andylokandy @kennytm @skyzh

File: prow/config/config.yaml [hunk 2]
```diff
@@ -1197,6 +1197,7 @@ tide:
         - do-not-merge/hold
         - do-not-merge/work-in-progress
         - do-not-merge/blocked-paths
+        - do-not-merge/invalid-commit-message
```
[the remark this answers is not in the document]
Comment (Mini256) [reply prow/config/config.yaml:1200]: When it is found that the repository does not have the label, the robot has sufficient authority to add the repository label automatically.
But you can add this label in advance and set its color.

**Review by rleungx — APPROVED** [review]

**Review by BusyJay — COMMENTED** [review]
LGTM for enabling the check.

**Pull request merged**